src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs

```diff
@@ -56,6 +56,96 @@ public class ResolveAssemblyReference : TaskExtension
         {
+            Strings.Initialize(Log);
+        }
+
+        private static class Strings
+        {
+            public const string FourSpaces = "    ";
+            public const string EightSpaces = "        ";
+            public const string TenSpaces = "          ";
+            public const string TwelveSpaces = "            ";
+
+            public static string ConsideredAndRejectedBecauseFusionNamesDidntMatch;
+            public static string ConsideredAndRejectedBecauseNoFile;
+            public static string ConsideredAndRejectedBecauseNotAFileNameOnDisk;
+            public static string ConsideredAndRejectedBecauseNotInGac;
+            public static string ConsideredAndRejectedBecauseTargetDidntHaveFusionName;
+            public static string Dependency;
+            public static string FormattedAssemblyInfo;
+            public static string FoundRelatedFile;
+            public static string FoundSatelliteFile;
+            public static string FoundScatterFile;
+            public static string ImageRuntimeVersion;
+            public static string IsAWinMdFile;
+            public static string LogAttributeFormat;
+            public static string LogTaskPropertyFormat;
+            public static string NoBecauseParentReferencesFoundInGac;
+            public static string NotCopyLocalBecauseConflictVictim;
+            public static string NotCopyLocalBecauseEmbedded;
+            public static string NotCopyLocalBecauseFrameworksFiles;
+            public static string NotCopyLocalBecauseIncomingItemAttributeOverrode;
+            public static string NotCopyLocalBecausePrerequisite;
+            public static string NotCopyLocalBecauseReferenceFoundInGAC;
+            public static string PrimaryReference;
+            public static string RemappedReference;
+            public static string RequiredBy;
+            public static string Resolved;
+            public static string ResolvedFrom;
+            public static string SearchedAssemblyFoldersEx;
+            public static string SearchPath;
+            public static string TargetedProcessorArchitectureDoesNotMatch;
+            public static string UnificationByAppConfig;
+            public static string UnificationByAutoUnify;
+            public static string UnificationByFrameworkRetarget;
+            public static string UnifiedDependency;
+            public static string UnifiedPrimaryReference;
+
+            internal static void Initialize(TaskLoggingHelper log)
+            {
+                if (Resolved != null)
+                {
+                    return;
+                }
+
+                string GetResource(string name) => log.GetResourceMessage(name);
+                string GetResourceFourSpaces(string name) => FourSpaces + log.GetResourceMessage(name);
+
+                ConsideredAndRejectedBecauseFusionNamesDidntMatch = EightSpaces + GetResource("ResolveAssemblyReference.ConsideredAndRejectedBecauseFusionNamesDidntMatch");
```

Comment:
> I'm particularly thinking of things like scatter files because I have no idea what they're for, and I asked rainersigwald at one point, and he had no idea what they were for. If no one knows what they're for, why are we logging about them? Are they even helpful, or can we remove them (and calculating them) entirely?

This sounds worth talking about when Rainer gets back. Maybe a changewave to log "less useful" messages at the lowest verbosity rather than not at all. Unless they're already logged at the lowest. Then I don't think we can avoid someone somewhere getting annoyed.

Reply:
Yes, it's tempting to reduce RAR logging. RAR logging is the thing that bloats the binlog size the most. We should hopefully have an effort to increase more useful logging and reduce useless logging. I couldn't resist and removed the "Image runtime is v4.0.30319" in this PR which is absolutely useless ("gee whiz, this .dll is a managed assembly").

But removing more should be a separate effort.

Reply:
In general, we should fight "Captain Obvious logging" everywhere and ensure that the signal-to-noise ratio is high. NuGet RestoreTask is especially high noise unfortunately: https://github.com/NuGet/Home/issues/10383